adal/build.gradle

```diff
@@ -67,6 +67,7 @@ android {
         disable 'GradleDependency'
         disable 'GradleOverrides'
         disable 'OldTargetApi'
+        disable 'PackageManagerGetSignatures'
```

Comment:
i suspect this could be happening because the method returns SignatureData and returns Null (2 different return statements) instead of one consolidates one. On a whim, can you try refactoring (locally) to combine both returns into one... (i.e. declare a variable with null to start with and then set to the correct value in for loop with just one return outside the loop) and then running the tool again to see if exception still occurs?

---
In reply to: [121008721](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/859#discussion_r121008721) [](ancestors = 121008721)